src/flexible_type/string_parser.hpp

```diff
@@ -229,10 +235,19 @@ struct string_parser
       if (!quote_char) boost::algorithm::trim_right(final_str);
       else if (quote_char) {
         // if was quoted field, we unescape the contents
-        turi::unescape_string(final_str, config.escape_char,
-                                  quote_char, config.double_quote);
+        turi::unescape_string(final_str, config.use_escape_char, 
+                              config.escape_char,
+                              quote_char, config.double_quote);
+      }
+      if (!config.na_val.empty() && config.na_val.count(final_str)) {
```

Comment:
Might be faster to just use a single hash map of key-to-value instead of 3 successive lookups in separate tables.